### PR TITLE
Implement Darwin v7 reader and improve annotation class

### PR DESCRIPTION
**Pull request by jonasteuwen**
The suggested changes have two major changes
- Labels in `WsiAnnotations` are now a class rather than a string. This means that you can have `AnnotationClass` with the label 'tumor' and the type being any AnnotationType
- Darwin V7 JSON 2.0 reader

We have also included the changes of #157

### CODE REVIEW

File: dlup/annotations.py [hunk 1]
```diff
@@ -495,30 +566,69 @@ def from_halo_xml(cls, halo_xml: PathLike, scaling: float | None = None) -> WsiA
...
+                x, y, h, w = curr_data.values()
+                curr_polygon = shapely.geometry.box(x, y, x + w, y + h)
+                curr_polygon = rescale_geometry(curr_polygon, scaling=scaling)
+                annotations[key].append(Polygon(curr_polygon, a_cls=_cls))
+            else:
+                ValueError(f"Annotation type {annotation_type} is not supported.")
+
+        # Now we can make WsiSingleLabel annotations
+        output = []
+        for an_cls in annotations:
```
Comment (martvanrijthoven) [inline dlup/annotations.py:626]:
```python
        for an_cls, coordinates in annotations.items():
            output.append(WsiSingleLabelAnnotation(a_cls=an_cls, coordinates=coordinates))
        return cls(output)
```

Comment (martvanrijthoven) [reply dlup/annotations.py:626]: Just a very small change to make it a bit more explicit that we get the "coordinates"

File: dlup/annotations.py [hunk 2]
```diff
@@ -855,6 +864,38 @@ def __str__(self):
...
 
+class DarwinPolygon:
+    def __init__(self, polygon):
+        self.geom = polygon
+        self.hole = False
+        self.holes = []
+
+
+def _parse_darwin_complex_polygon(annotation):
+    polygons = [DarwinPolygon(Polygon([(p["x"], p["y"]) for p in path])) for path in annotation.data["paths"]]
```
Comment (martvanrijthoven) [inline dlup/annotations.py]: annotation.data["paths" --> annotation["paths"]

File: dlup/annotations.py (hunk 1 above)
Comment (jonasteuwen) [reply dlup/annotations.py]: Done

**Comment by martvanrijthoven** [comment]
Looks good :). Here is an test json file for testing darwin parsing:

[103S.json.zip](https://github.com/NKI-AI/dlup/files/12370603/103S.json.zip)

**Review by martvanrijthoven — APPROVED** [review]
It looks to me that this is ready to go :)

**Pull request merged**